src/audio/pipeline/pipeline-stream.c

```diff
@@ -430,7 +433,7 @@ int pipeline_trigger_run(struct pipeline *p, struct comp_dev *host, int cmd)
 		list_init(&walk_ctx.pipelines);
 
 		if (data.delay_ms)
-			wait_delay_ms(data.delay_ms);
+			k_msleep(data.delay_ms);
```

Comment:
note, that this might incur a context switch now, IIUC. Before it was a busy loop, now under Zephyr this can actually cause a context switch. Although this path is only taken for DMA-driven pipelines and only when that delay is non-zero... And there might not be any other threads to switch to. So in practice this should be harmless and have no effect in 99.99% of the cases

Reply:
yep, this is intended - and we do need to rework the DMA driver scheduling at some point.